Add EFP data to Address Book

encryptedDegen opened the pull request:
This is a PR realted to the issue https://github.com/superfluid-org/superfluid-dashboard/issues/787 to add the Ethereum Follow Protocol data to the Address Book as a separate table.

This PR introduces:
- efpApi slice for fetching user#s Followings (paginated) and stats
- populating a new table with the data from the efp api
- empty table that links to EFP so they can start following people to then get displayed in Superfluid address book

encryptedDegen committed:
fetch EFP data and display in address book

encryptedDegen committed:
revert reformatting of the store

encryptedDegen committed:
change following order, beautify ENS names, add empty table

encryptedDegen commented:
I have noticed you are not beautifying the ENS names with the function you have available, so I have added it in the Address Book page in the tables. It correctly interprets the emojis which can usually appear vroken after resolving the ENS name.

Before:
![image](https://github.com/user-attachments/assets/54a09c24-3b65-49a9-a3cd-9c46625bcb11)

After
![image](https://github.com/user-attachments/assets/20e71da3-16ee-4334-9ee6-decbbf0a92f2)
![image](https://github.com/user-attachments/assets/e6d10fe9-a82d-4735-a38c-25210e6dc6cf)

encryptedDegen committed:
handle disconnected following table

elvijsTDL reviewed (changes requested):
Hey @encryptedDegen , checked the changes and I can see it picks up the followings in the address book ,but the problem I have with this is that it does not work as an address book entry, the main idea is that once you have someone added to address book it shows up in the list in the receiver menus all over the page 

As you can see in this example , if I start to write "te" it already shows all the address book entries I would have with "te" in them  , if the EFP followings don't work in this way , it makes the user basicly look up the ENS name again , making this feature not really that useful 
![Screenshot 2025-03-20 at 14 36 43](https://github.com/user-attachments/assets/6a52d752-7c39-4166-a876-afed78c6045f)

It would also be great if you could show a loading skeleton in the address book page while fetching the followings, instead of showing the message as if there were no accounts to follow , other than that LGTM

encryptedDegen commented:
> Hey @encryptedDegen , checked the changes and I can see it picks up the followings in the address book ,but the problem I have with this is that it does not work as an address book entry, the main idea is that once you have someone added to address book it shows up in the list in the receiver menus all over the page
> 
> As you can see in this example , if I start to write "te" it already shows all the address book entries I would have with "te" in them , if the EFP followings don't work in this way , it makes the user basicly look up the ENS name again , making this feature not really that useful ![Screenshot 2025-03-20 at 14 36 43](https://private-user-images.githubusercontent.com/77115130/424981030-6a52d752-7c39-4166-a876-afed78c6045f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI0NzgxNDgsIm5iZiI6MTc0MjQ3Nzg0OCwicGF0aCI6Ii83NzExNTEzMC80MjQ5ODEwMzAtNmE1MmQ3NTItN2MzOS00MTY2LWE4NzYtYWZlZDc4YzYwNDVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIwVDEzMzcyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZiZGRjM2NkNmY0ZDE1ZGY5NjA5ZTk3MDk3OGVjNTE2MjI3NjAzNmRlYjQxYzMxZGY1NDVhMTU2OTY3N2M2YzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aP_6o8cn6dN8vRhsz3nxpbtEcWD8aKm_CVRiW4eqpdY)
> 
> It would also be great if you could show a loading skeleton in the address book page while fetching the followings, instead of showing the message as if there were no accounts to follow , other than that LGTM

Hey @elvijsTDL, thanks for the feedback.

I completely agree with you. I will add support to searchbar to fetch EFP followings that match the query string and the loading skeleton for the table in the address book.

Will ping you as soon as it's done.

encryptedDegen committed:
add loading rows to followings table, add EFP data to address search dialog

encryptedDegen commented:
@elvijsTDL I have performed the requested changes.

Address book now has loading rows for desktop & mobile (I made them a separate components to not mess with any states inside the original components)
Added EFP following data to Address Search Dialog component (limited to 10 results for faster load times and to not flood the results if only one letter is typed)

I hope the PR now includes all the desired functionality and is up to the code standards.

encryptedDegen committed:
add favouriting functionality to EFP table in address book

encryptedDegen commented:
@elvijsTDL I have added another little piece that was missing from the mockup we have provided https://github.com/superfluid-org/superfluid-dashboard/issues/787#issue-2892141571.

Since EFP followings are meant to complement the address book functionality, I have added the stars to the EFP following table rows where you can easily add the person to your address book for easier access across the app. Works like this:
https://github.com/user-attachments/assets/7d97115d-514f-48ed-8ead-51f2d4a48529

Should enhance the UX since users won't always need to go through multiple pages of followings, but have their favourites in the address book with a single click.

caveman-eth commented:
We have identified one bug in this preview, where a favorite address is not showing when you type the ENS in the send feature.
We will fix this quickly and get back to you.

caveman-eth commented:
@elvijsTDL @kasparkallas I think we are ready for review now :)

You can disregard my previous comment, I was mistakenly thinking we had introduced a bug, but I see now that it is normal behavior for an address without a nickname.

Let us know if you want anymore changes, or if you think anything else is missing that you would like added, will be happy to address.
Thanks for your patience and feedback 👍

hellwolf commented:
@kasparkallas let's merge?

kasparkallas reviewed (approved):
LGTM 💚

Good job figuring out all the redux stuff!
